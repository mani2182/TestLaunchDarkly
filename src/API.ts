// API request Methods for launchDarkly
import axios from 'axios';
import {getBaseURI} from './constants';

export const CallAPI = async (
  method: string,
  body: string | object | null | '',
  tokenId: string,
) => {
  const getURL = getBaseURI + tokenId + '/contexts/' + body; //launckDarkly URL
  let setApiData = {};
  switch (method) {
    case 'GET':
      try {
        let response = await axios
          .get(getURL)
          .then((resp: any) => {
            return resp.data;
          })
          .catch((error: any) => {
            return error.response.data;
          });
        setApiData = response;
      } catch (error: any) {
        setApiData = {type: 'error', message: 'An unexpected error occurred'};
      }
      break;
    default:
  }
  return setApiData;
};
