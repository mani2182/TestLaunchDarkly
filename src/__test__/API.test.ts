import axios from 'axios';
import { CallAPI } from '../API';

describe('CallAPI function', () => {
  it('handles successful GET API call', async () => {
    const mockedResponse = { data: 'Mocked API response' };
    const axiosGetSpy = jest
      .spyOn(axios, 'get')
      .mockResolvedValue({ data: mockedResponse });

    const result = await CallAPI('GET', 'mockBody', 'mockToken');

    expect(result).toEqual(mockedResponse);
    expect(axiosGetSpy).toHaveBeenCalledWith(
      'https://app.launchdarkly.com/sdk/evalx/mockToken/contexts/mockBody',
    );
  });

  it('handles API error on GET', async () => {
    const errorMessage = 'API error message';
    const axiosGetSpy = jest
      .spyOn(axios, 'get')
      .mockRejectedValue({ response: { data: errorMessage } });

    const result = await CallAPI('GET', 'mockBody', 'mockToken');

    expect(result).toBeDefined();
    expect(axiosGetSpy).toHaveBeenCalledWith(
      'https://app.launchdarkly.com/sdk/evalx/mockToken/contexts/mockBody',
    );
  });

  it('handles unexpected error', async () => {
    const axiosGetSpy = jest
      .spyOn(axios, 'get')
      .mockRejectedValue(new Error('Unexpected error'));

    const result = await CallAPI('GET', 'mockBody', 'mockToken');

    expect(result).toEqual({
      type: 'error',
      message: 'An unexpected error occurred',
    });
    expect(axiosGetSpy).toHaveBeenCalledWith(
      'https://app.launchdarkly.com/sdk/evalx/mockToken/contexts/mockBody',
    );
  });

  it('handles unknown method', async () => {
    const tokenId = 'sample-token';
    const body = 'sample-body';
    const method = 'UNKNOWN';
    const response = await CallAPI(method, body, tokenId);
    expect(response).toEqual({});
  });
});

describe('snapshot for API component', () => {
  it('renders snapshot without issues', () => {
    expect(CallAPI).toMatchSnapshot();
  });
});