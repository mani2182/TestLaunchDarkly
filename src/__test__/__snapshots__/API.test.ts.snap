// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot for API component renders snapshot without issues 1`] = `[Function]`;
